packages/embark-transaction-manager/src/lib/eth_accounts.ts

```diff
@@ -0,0 +1,36 @@
+import { Callback, Embark, Logger } /* supplied by @types/embark in packages/embark-typings */ from "embark";
+import Web3 from "web3";
+const { blockchain: blockchainConstants } = require("embark-core/constants");
+import { __ } from "embark-i18n";
+
+export default class EthAccounts {
+  private readonly logger: Logger;
+  constructor(embark: Embark, web3: Web3, private readonly accounts: any[]) {
+    this.logger = embark.logger;
+
+    embark.registerActionForEvent("blockchain:proxy:response", this.checkResponseFor_eth_accounts.bind(this));
+  }
+
+  private async checkResponseFor_eth_accounts(params: any, callback: Callback<any>) {
+    const txsToModify = [
+      blockchainConstants.transactionMethods.eth_accounts,
+      blockchainConstants.transactionMethods.personal_listAccounts,
+    ];
+
+    if (!(txsToModify.some((txMethod) => txMethod === params.reqData.method) && this.accounts.length)) {
+      return callback(null, params);
+    }
+
+    this.logger.trace(__(`Modifying blockchain '${params.reqData.method}' response:`));
+    this.logger.trace(__(`Original request/response data: ${JSON.stringify(params)}`));
+
+    try {
+      params.respData.result = this.accounts.map((acc) => acc.address || acc);
```

Comment:
Great point. I didn't actually think of node accounts changing. I added a tx interception for `personal_newAccount`, which then informs the transaction-manager (via a private event bus), and all the transaction modifiers will have their account modified accordingly. This will handle cases when a user adds a new account via `web3.personal.newAccount` API. 

For the lines highlighted above, it seems this would handle the case if the user adds a node account to the node itself, and then a call is made to `eth_accounts`. I added back some of the code to modify the response in that case, but also emits an event so that all other tx modifiers can have their node accounts updated as well.